engine/src/main/java/io/camunda/zeebe/engine/state/analyzers/CatchEventAnalyzer.java

```diff
@@ -135,13 +146,103 @@ private Either<List<DirectBuffer>, CatchEventTuple> findCatchEventInScope(
   }
 
   public Optional<CatchEventTuple> findEscalationCatchEvent(
-      final DirectBuffer escalationCode, final ElementInstance instance) {
-    // TODO walk through the scope hierarchy and look for a matching catch event
+      final DirectBuffer escalationCode, ElementInstance instance) {
+    // walk through the scope hierarchy and look for a matching catch event
+    while (instance != null && instance.isActive()) {
+      final var instanceRecord = instance.getValue();
+      final var process = getProcess(instanceRecord.getProcessDefinitionKey());
+
+      final var found = findEscalationCatchEventInProcess(escalationCode, process, instance);
+      if (found.isPresent()) {
+        return found;
+      }
+
+      // find in parent process instance if exists
+      final var parentElementInstanceKey = instanceRecord.getParentElementInstanceKey();
+      instance = elementInstanceState.getInstance(parentElementInstanceKey);
+    }
+
+    // no matching catch event found
+    return Optional.empty();
+  }
+
+  private Optional<CatchEventTuple> findEscalationCatchEventInProcess(
+      final DirectBuffer escalationCode,
+      final ExecutableProcess process,
+      ElementInstance instance) {
+
+    while (instance != null && instance.isActive() && !instance.isInterrupted()) {
+      final var found = findEscalationCatchEventInScope(escalationCode, process, instance);
+      if (found.isPresent()) {
+        return found;
+      }
+      // find in parent scope if exists
+      final var instanceParentKey = instance.getParentKey();
+      instance = elementInstanceState.getInstance(instanceParentKey);
+    }
 
     // no matching catch event found
     return Optional.empty();
   }
 
+  private Optional<CatchEventTuple> findEscalationCatchEventInScope(
+      final DirectBuffer escalationCode,
+      final ExecutableProcess process,
+      final ElementInstance instance) {
+    final var processInstanceRecord = instance.getValue();
+    final var elementId = processInstanceRecord.getElementIdBuffer();
+    final var elementType = processInstanceRecord.getBpmnElementType();
+
+    final var element = process.getElementById(elementId, elementType, ExecutableActivity.class);
+    if (element.getEvents().isEmpty()) {
+      return Optional.empty();
```

Comment:
❌ We should change this method to not care about whether or not an event is a start event or a boundary event. We can quite easily achieve this with a simple stream. Using this stream would make this check obsolete so we can remove it.